src/Compilers/CSharp/Portable/BoundTree/Constructors.cs

```diff
@@ -25,8 +25,9 @@ internal sealed partial class BoundFieldAccess
             ConstantValue constantValueOpt,
             LookupResultKind resultKind,
             TypeSymbol type,
-            bool hasErrors = false)
-            : this(syntax, receiver, fieldSymbol, constantValueOpt, resultKind, NeedsByValueFieldAccess(receiver, fieldSymbol), type, hasErrors)
+            bool hasErrors = false,
+            bool isDeclaration = false)
```

Comment:
>bool isDeclaration = false [](start = 12, length = 26)

This feels error-prone, two booleans at the end and hasErrors is not the last one, this goes against the common pattern. If the number of call-sites that use isDeclaration == true small, perhaps we can simply use full constructor there.

Reply:
I'll introduce a separate constructor, since this constructor uses a private method I can't get access to outside.

Reply:
Fixed.